src/engine/SCons/Node/__init__.py

```diff
@@ -137,6 +137,8 @@ def exists_always(node):
     return 1
 
 def exists_base(node):
+    if node.islink():
```

Comment:
can we move this logic into FS.Base.stat()?

Reply:
You could but I would think that it would be weird that the `stat` method no longer corresponds to the stat syscall. It would then be doing something more akin to `exists`.

I'm happy to make that change if you think that's right

Reply:
Could this be simplifed ot skip the islink() like this:

```
def exists_base(node):
    return node.lstat() is not None and node.stat() is not None 
```

It looks like lstat() on a non symlink returns same info as stat() so worst case you effectively ask for stat twice?

From python 3.5 docs

> On platforms that do not support symbolic links, this is an alias for stat().
> 
> As of Python 3.3, this is equivalent to os.stat(path, dir_fd=dir_fd, follow_symlinks=False).

Reply:
I don't think exists_base needs to be changed at all here based off a small table I made, you'll notice the results are the same for `(stat is not None)` and `(stat is not None and lstat is not None)`:

  | symlink exists, file exists | symlink exists, file doesnt exists | symlink doesn't exist, file exists | symlink doesn't exist, file doesnt exist
-- | -- | -- | -- | --
stat is not None | TRUE | FALSE | FALSE | FALSE
lstat is not None | TRUE | TRUE | FALSE | FALSE
stat is not None and lstat is not None | TRUE | FALSE | FALSE | FALSE

Reply:
@dmoody256  did you try linux, windows, macos?

Reply:
I tested manually on linux, and for windows, lstat would become the same as stat according to the docs you quoted. In my follow on [PR](https://github.com/SCons/scons/pull/3638), the tests introduced in this PR ran and passed on windows and linux with no change to the exists_base function.

So really macos is the only one untested, and I am assuming its the same as other *nix platforms for symlinking. Someone correct me if I'm wrong.